From 0f06280595eca85c6f7301e455f6a3e0c242a0ad Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Wed, 7 Jun 2023 17:11:34 +0200
Subject: [PATCH 0351/1141] dt-bindings: dma: stm32-dma3: introduce
 memory-region property

DMA3 requires memory to allocate its linked-list items (LLIs). By default,
when memory-region property is not specified, LLIs are allocated in DDR.
Thanks to memory-region property, a dedicated /reserved-memory child node
can be used for LLIs allocation. It is then recommended to use internal
memory: LLI read for registers update is optimal from internal RAM.

Change-Id: I59bb69a0abbb3f8cb9b83801730de35daddd63e3
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/312803
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Antonio Maria BORNEO <antonio.borneo@foss.st.com>
---
 Documentation/devicetree/bindings/dma/st,stm32-dma3.yaml | 9 +++++++++
 1 file changed, 9 insertions(+)

--- a/Documentation/devicetree/bindings/dma/st,stm32-dma3.yaml
+++ b/Documentation/devicetree/bindings/dma/st,stm32-dma3.yaml
@@ -103,6 +103,15 @@ properties:
     minItems: 4
     maxItems: 16
 
+  memory-region:
+    description: |
+      Phandle to a /reserved-memory child node assigned to STM32 DMA3 for linked-list items
+      allocation. When not defined, linked-list items are allocated in DDR.
+      During transfer, STM32 DMA3 updates its registers reading the linked-list items in memory.
+      There reads are 32-bit single read. Allocating the linked-list items in internal RAM improves
+      the STM32 DMA3 performance.
+    maxItems: 1
+
   st,axi-max-burst-len:
     description: |
       Restrict master AXI burst length in unit of beat by value specified in this property.
